File: java/src/test/java/io/cucumber/cucumberexpressions/ArgumentTest.java
package io.cucumber.cucumberexpressions;

import org.junit.jupiter.api.Test;

import java.util.List;
import java.util.Locale;

import static java.util.Collections.singletonList;
import static java.util.Objects.requireNonNull;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class ArgumentTest {
    @Test
    public void exposes_parameter_type() {
        TreeRegexp treeRegexp = new TreeRegexp("three (.*) mice");
        Group group = requireNonNull(treeRegexp.match("three blind mice"));
        
        ParameterTypeRegistry parameterTypeRegistry = new ParameterTypeRegistry(Locale.ENGLISH);
        List<ParameterType<?>> parameterTypes = singletonList(parameterTypeRegistry.lookupByTypeName("string"));
        
        List<Argument<?>> arguments = Argument.build(group, parameterTypes);
        Argument<?> argument = arguments.get(0);
        assertEquals("string", argument.getParameterType().getName());
    }

}

File: elixir/test/varar/cucumber_expressions/argument_test.exs
defmodule Varar.CucumberExpressions.ArgumentTest do
  use ExUnit.Case, async: true

  alias Varar.CucumberExpressions.{Argument, ParameterTypeRegistry, TreeRegexp}

  test "exposes parameter_type" do
    tree_regexp = TreeRegexp.new!(~r/three (.*) mice/)
    registry = ParameterTypeRegistry.new()
    string_type = ParameterTypeRegistry.lookup_by_type_name(registry, "string")

    [argument] = Argument.build(tree_regexp, "three blind mice", [string_type])
    assert argument.parameter_type.name == "string"
  end
end
